modules/gdscript/gdscript_parser.cpp

```diff
@@ -4345,71 +4345,81 @@ void GDScriptParser::_parse_class(ClassNode *p_class) {
 						}
 						parenthesis--;
 
-						if (subexpr->type != Node::TYPE_CONSTANT) {
-							current_export = PropertyInfo();
-							_set_error("Expected a constant expression.");
-						}
+						if (subexpr->type == Node::TYPE_IDENTIFIER) {
```

Comment:
Shouldn't there be some kind of check here to confirm that the specified Identifier is a name within the ScriptServer, or would that cause scenario collision with the GDScriptNativeClass? I would expect me using an invalid script class name here would trigger a `_set_error` call (haven't tested this yet though).